bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/QuantityTypeArithmeticGroupFunction.java

```diff
@@ -62,208 +61,148 @@ public State[] getParameters() {
             return new State[0];
         }
 
-        protected boolean isSameDimension(@Nullable Item item) {
-            if (item instanceof GroupItem groupItem) {
-                return isSameDimension(groupItem.getBaseItem());
-            }
-            return item instanceof NumberItem ni && dimension.equals(ni.getDimension());
+        /**
+         * Convert the given item {@link State} to a {@link QuantityType} based on the {@link Unit} of the
+         * {@link GroupItem} i.e. 'referenceUnit'. Returns null if the {@link State} is not a {@link QuantityType} or
+         * if the {@link QuantityType} could not be converted to 'referenceUnit'.
+         *
+         * The conversion can be made to both inverted and non-inverted units, so invertible type conversions (e.g.
+         * Mirek <=> Kelvin) are supported.
+         *
+         * @param state the State of any given group member item
+         * @return a QuantityType or null
+         */
+        private @Nullable QuantityType<?> referenceUnitQuantityType(@Nullable State state) {
+            return state instanceof QuantityType<?> quantity ? quantity.toInvertibleUnit(referenceUnit) : null;
+        }
+
+        /**
+         * Convert a set of {@link Item} to a respective list of {@link QuantityType}. Exclude any {@link Item}s whose
+         * current {@link State} is not a {@link QuantityType}. Convert any remaining {@link QuantityType} to the
+         * 'referenceUnit' and exclude any values that did not convert.
+         *
+         * @param items a list of {@link Item}
+         * @return a list of {@link QuantityType} converted to the 'referenceUnit'
+         */
+        @SuppressWarnings({ "rawtypes" })
+        protected List<QuantityType> referenceUnitQuantityTypes(Set<Item> items) {
+            return items.stream().map(i -> i.getStateAs(QuantityType.class)).map(s -> referenceUnitQuantityType(s))
+                    .filter(Objects::nonNull).map(s -> (QuantityType) s).toList();
```

Comment:
```suggestion
    protected List<QuantityType<?>> referenceUnitQuantityTypes(Set<Item> items) {
        return items.stream().map(i -> i.getState()).map(s -> referenceUnitQuantityType(s))
                .filter(Objects::nonNull).collect(Collectors.toList());
    }
```
I think this solves the rawtypes problem. I still had it when using a straight `toList()`.

Reply:
> I think this solves the rawtypes problem 

Umm. Yes it solves the problem on `toList()`. But the method still needs the annotation for another reason. Aargh!

Reply:
Yes, figured it out as well. That's where I find streams combined with generics are a huge challenge. See my updated comment.

Reply:
> You can just have i.getState()

Good catch. Done.

I think the remaining issues boil down to the different generics signatures between `QuantityTypeArithmeticGroupFunction.java` (i.e. `QuantityType<T>`) and `QuantityType.java` (i.e. `QuantityType<?>` .. the former is what we get from the stream functions, and latter is what we need when we call `QuantityType.add()`..

I did not fully follow you PR discussion about `compareTo` yet .. but I saw that `QuantityType.compareTo(QuantityType<T> o)` has a different signature than `QuantityType.internalCompareTo(QuantityType<?> o)` .. so I wonder if we could do the same (actually the opposite) here? And if it is actually necessary?